File: cash.c
#include <cs50.h>
#include <stdio.h>

int get_cents(void);
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void) {
    int cents = get_cents();

    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;

    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;

    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;

    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;
    
    int coins = quarters + dimes + nickels + pennies;
    printf("%i\n", coins);
}

int get_cents(void) {
    int cents;
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);

    return cents;
}

int calculate_quarters(int cents) {
    int n;
    n = cents / 25;

    return n;
}

int calculate_dimes(int cents) {
    int n;
    n = cents / 10;

    return n;
}

int calculate_nickels(int cents) { 
    int n;
    n = cents / 5;

    return n;
}

int calculate_pennies(int cents) {

    return cents;
}

File: week6/sentimental-cash/cash.py
def get_dollars():
    while True:
        try:
            dollars = float(input("Change owed: "))
            if (dollars > 0):
                break
        except ValueError:
            print("Thats not a float!")

    return dollars
        
def calculate_quarters(dollars):
    return int((dollars * 100.0) / 25.0)

def calculate_dimes(dollars):
    return int((dollars * 100.0) / 10.0)

def calculate_nickels(dollars):
    return int((dollars * 100.0) / 5.0)

def calculate_pennies(dollars):
    return int(dollars * 100)

def main():
    dollars = get_dollars()

    quarters = calculate_quarters(dollars)
    dollars = (dollars * 100.0 - quarters * 25.0) / 100.0

    dimes = calculate_dimes(dollars)
    dollars = (dollars * 100.0 - dimes * 10.0) / 100.0

    nickels = calculate_nickels(dollars)
    dollars = (dollars * 100.0 - nickels * 5.0) / 100.0

    pennies = calculate_pennies(dollars)
    dollars = (dollars * 100.0 - pennies) / 100.0

    coins = quarters + dimes + nickels + pennies
    coins = int(coins)

    print(coins)
    print("\n", end="")

main()
